tmt/container/__init__.py

```diff
@@ -964,6 +964,14 @@ def from_yaml(cls, yaml: str) -> Self:
             return cls.model_validate(tmt.utils.yaml_to_dict(yaml))
 
         except ValidationError as error:
-            import tmt.utils
-
             raise tmt.utils.SpecificationError("Invalid metadata in YAML data.") from error
+
+    @classmethod
+    def from_json(cls, json_data: str) -> Self:
+        import tmt.utils
+
+        try:
+            return cls.model_validate_json(json_data)
```

Comment:
It turns out that `model_validate_json()` is not available with Pydantic 1.x. We have two options: provide a shim in `tmt._compat.pydantic.BaseModel`, or modify this method to use `json.loads()` + `model_validate()`, just like `from_yaml()` does. The former might be better for the overall usability of our models.